hw/ip/pwrmgr/doc/_index.md

```diff
@@ -0,0 +1,233 @@
+The power manager sequences the design from a freshly reset state to an active state where software can be initialized.
+
+Low power entry is the process in which the device enters one of two low power modes (sleep or deep sleep).
+Low power exit is the process in which the device exits low power mode and returns to active state.
+Low power entry is always initiated by software, while low power exit is always initiated by a previously setup hardware event such as pins or internal timers.
+The power manager processes the software and hardware requests to perform the appropriate actions.
+
+Reset scenarios refer to non-por events that cause the device to reboot.
+There are various stimuls that can cause such a reset, ranging from exteranl user input to watchdog.
+The power manager processes the reset request and brings the device to an appropriate state.
+
+
+# Theory Of Operation
+
+The power manager controls power domain(s) turn on / turn off, the root resets in conjunction with the reset manager, the root clock enables in conjunction with AST / clock manager and various power up activities such as OTP sensing and life cycle initiation before allowing software to execute.
+
+In the future, if there is a need to perform ROM hashing as an integrity check, that will also be managed by the power manager.
+
+## Block Diagram
+
+See the below high level block diagram that illustrates the connections between the power manager and various system components.
+Blocks outlined with a solid magenta line are always on; while blocks outlined with a dash magenta line are a mix of always-on and non-always-on components.
+
+![Power Manager Connectivity Diagram](pwrmgr_connectivity.svg)
+
+
+## Overall Sequencing
+
+The power manager primarily is composed of two state machines.
+One operates on the always-on slow clock (this clock is always running and usually measured in KHz) and is responsible for turning faster clocks on and off and managing the power domains.
+The other operates on a normal fixed clock (usually measured in MHz) and is responsible for everything else in the power sequence.
+
+The following diagram breaks down the general functionality of both.
+
+
+![Power Manager FSMs](pwrmgr_fsms.svg)
+
+
+Note, most of the states are transitional states, and only the following state combinations are resting states.
+
+
+*   Slow FSM IDLE and fast FSM ACTIVE
+*   Slow FSM LOW\_POWER and fast FSM LOW\_POWER
+
+The slow FSM LOW\_POWER and fast FSM ACTIVE states specifically are concepts useful when examining [reset handling](#Reset-Request-Handling).
+
+
+## Slow Clock Domain FSM
+
+The slow clock domain FSM (referred to as the slow FSM from here on) resets to the RESET state.
+This state is released by por\_rst\_n, which is supplied from the reset controller.
+Por\_rst\_n releases when the reset controller detects the root power domains (aon_pok) of the system are ready.
+Please see the [AST interface]() for more details.
```

Comment:
ah this is because the AST documentation PR has not been merged yet.  Once it is, I'm going to reference there.
I didn't want to put the link down yet because it would be broken. (although I guess it's broken now, so let me just remove it and re-introduce it later).

Reply:
Cool, just wanted to check this wasn't a typo.